nexus/db-model/src/unsigned.rs

```diff
@@ -0,0 +1,113 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use diesel::backend::{Backend, RawValue};
+use diesel::deserialize::{self, FromSql};
+use diesel::pg::Pg;
+use diesel::serialize::{self, ToSql};
+use diesel::sql_types;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+/// Representation of a [`u16`] in the database.
+/// We need this because the database does not support unsigned types.
+/// This handles converting from the database's INT4 to the actual u16.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    AsExpression,
+    Eq,
+    Ord,
+    PartialEq,
+    PartialOrd,
+    FromSqlRow,
+    Serialize,
+    Deserialize,
+)]
+#[diesel(sql_type = sql_types::Int4)]
```

Comment:
Should you also delete the `u16.rs` file given that it's defined here now?

Reply:
Removed